Library/Homebrew/software_spec.rb

```diff
@@ -402,20 +403,28 @@ def compatible_locations?
   # Does the {Bottle} this {BottleSpecification} belongs to need to be relocated?
   sig { returns(T::Boolean) }
   def skip_relocation?
-    cellar == :any_skip_relocation
+    bottle_checksum = checksum_for(Utils::Bottles.tag).first
+    bottle_checksum.cellar == :any_skip_relocation
   end
 
   def tag?(tag)
     checksum_for(tag) ? true : false
   end
 
-  # Checksum methods in the DSL's bottle block optionally take
+  # BottleChecksum methods in the DSL's bottle block take
   # a Hash, which indicates the platform the checksum applies on.
-  Checksum::TYPES.each do |cksum|
-    define_method(cksum) do |val|
-      digest, tag = val.shift
-      collector[tag] = Checksum.new(cksum, digest)
-    end
+  # Example args:
+  # {"22948764d8f8d7be4870ff92dae64d986eb63a9150b219c20fff87d1a6aa93d6"=>{"tag"=>:mojave, "cellar"=>:any_skip_relocation}}
+  define_method(BottleChecksum::TYPE) do |*args|
```

Comment:
Given this is only ever `sha256` this feels like it might be a bit overly generic.

Reply:
Done